v2/internal/testcommon/vcr/v3/tracking_roundtripper.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package v3
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"io"
+	"net/http"
+	"sync"
+
+	"github.com/Azure/azure-service-operator/v2/internal/testcommon/vcr"
+)
+
+// Wraps an inner HTTP roundtripper to add a
+// counter for duplicated request URIs. This
+// is then used to match up requests in the recorder
+// - it is needed as we have multiple requests with
+// the same Request URL and it will return the first
+// one that matches.
+type requestCounter struct {
+	inner http.RoundTripper
+
+	countsMutex sync.Mutex
+	counts      map[string]uint32
+}
+
+func AddTrackingHeaders(inner http.RoundTripper) *requestCounter {
+	return &requestCounter{
+		inner:       inner,
+		counts:      make(map[string]uint32),
+		countsMutex: sync.Mutex{},
+	}
+}
+
+const (
+	COUNT_HEADER string = "TEST-REQUEST-ATTEMPT"
+	HASH_HEADER  string = "TEST-REQUEST-HASH"
+)
+
+var _ http.RoundTripper = &requestCounter{}
+
+func (rt *requestCounter) RoundTrip(req *http.Request) (*http.Response, error) {
+	if req.Method == "PUT" || req.Method == "POST" {
+		rt.addHashHeader(req)
```

Comment:
minor: This adds a hash _and_ a count header. Consider reflecting that reality in the method name, either by saying `HashAndCount` or by saying something like `addTrackingHeaders`

Reply:
Did you miss this comment?

Reply:
Had trouble coming up with a non-misleading name that was better than that. But I did, now updated.